scope.ml

```diff
@@ -24,24 +36,29 @@ let infer ({formals; instrs} : analysis_input) : inferred_scope array =
 
   let infer_scope instructions =
     let merge pc cur incom =
-      if not (VarSet.equal cur.info incom.info)
+      if not (DeclSet.equal cur.info incom.info)
       then raise (IncompatibleScope (cur, incom, pc))
       else if PcSet.equal cur.sources incom.sources then None
       else Some { info = cur.info; sources = PcSet.union cur.sources incom.sources }
     in
     let update pc cur =
       let instr = instructions.(pc) in
-      let added = Instr.declared_vars instr in
+      let declared = Instr.declared_vars instr in
+      let declared' = VarSet.elements declared in
+      let declared' = List.map (fun v -> (v, pc)) declared' in
+      let declared' = DeclSet.of_list declared' in
       let info = cur.info in
-      let shadowed = VarSet.inter info added in
+      let shadowed = VarSet.inter (decl_as_var_set info) declared in
```

Comment:
It seems this is the only other place where `declared` is used. Maybe you could write this as
```ocaml
let shadowed = decl_as_var_set (DeclSet.inter info declared')
```
and then get rid of `declared`?